src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticAnalysisContextHelpers.cs

```diff
@@ -158,7 +162,7 @@ private static void VerifyOperationKinds(ImmutableArray<OperationKind> operation
             }
         }
 
-        internal static void VerifyArguments<TKey, TValue>(TKey key, AnalysisValueProvider<TKey, TValue> valueProvider)
+        internal static void VerifyArguments<TKey, TValue>(TKey key, AnalysisValueProvider<TKey, TValue>? valueProvider)
```

Comment:
Given that the method throws when `valueProvider` is null, we should probably annotate that parameter as not-nullable.